app/controllers/locations_controller.rb

```diff
@@ -33,7 +33,13 @@ def create
     respond_to do |format|
       if @location.save
         format.html { redirect_to @location, notice: 'Location was successfully created.' }
-        format.json { render :show, status: :created, location: @location }
+        format.json do
+          if params[:source] == 'volunteers-form'
```

Comment:
I am not familiar with how the other json request worked, so added a source param to return json, instead of rendering show template.